pkgs/tools/misc/calamares/0004-Adds-unfree-qml-to-packagechooserq.patch

```diff
@@ -78,7 +78,7 @@ index 000000000..5e36d77d9
 +                    width: 600
 +                    height: 150
 +                    anchors.centerIn: parent
-+                    text: qsTr("NixOS is fully open source, but it also provides software packages with unfree licenses. By default unfree packages are not allowed, but you can enable it here. If you check this box, software installed might have additional End User License Agreements (EULAs) attached. If not enabled, some hardware might not work fully when no suitable open source drivers are available.<br/>")
++                    text: qsTr("NixOS is fully open source, but it also provides optional software packages that do not respect user freedom and are not open source. By default such unfree packages are not allowed, but you can enable it here. If you check this box, you agree that unfree software may be installed which might have additional End User License Agreements (EULAs) that you need to agree to. If not enabled, some hardware (particularly Nvidia GPUs and some WiFi chips) might not work or not work optimally.<br/>")
```

Comment:
> that do not respect user freedom and are not open source

This isn't really a factual statement. Unfree license does not mean it's not open source. There are plenty of open source licenses and projects that violate some aspect of the four freedoms. Hell, there are debates about which licenses do and don't violate these freedoms. I think it's better to leave interpretation out of it and just leave it as "with unfree licenses". I think we can respect our users' intelligence and leave it at that.

> it here. If you check this box, you agree that unfree software may be installed which might have additional End User License Agreements (EULAs) that you need to agree to

Frankly, this whole sentence was not the right thing to say in the first place. I don't think the installer installs unfree software *at all*, even with this checked. It should say that checking this box *allows you to add unfree software of your choice after installation*, and perhaps that it is up to you to determine the license restrictions of that software.

> If not enabled, some hardware (particularly Nvidia GPUs and some WiFi chips) might not work or not work optimally.

I don't like the word "particularly" here. To me, this word implies that these are the specific things one needs to consider, and not anything else except in esoteric cases. So I would use wording indicating that these are *common* instances, like "for instance" or something.

Reply:
> I don't think the installer installs unfree software _at all_, even with this checked. It should say that checking this box _allows you to add unfree software of your choice after installation_, and perhaps that it is up to you to determine the license restrictions of that software.

I'm not sure whether that's the case.

For some context: I just saw this screen in a screenshot and thought to myself: "Hm, that's a little unclear.". I've never used the graphical installer. I can imagine it might perhaps add proprietary drivers if detected though? Perhaps @vlinkz could share some insight here.

Reply:
> Unfree license does not mean it's not open source. There are plenty of open source licenses and projects that violate some aspect of the four freedoms.

I‘m confused – do you mean “source‐available” licenses? It’s generally widely agreed that “open source” and “Free software” are essentially synonyms modulo political connotations (hence terms like “FOSS”), and the “open source” terminology was popularized by the Open Source Initiative, which is generally considered one of the canonical sources for FOSS licence approvals and whose [Open Source Definition](https://opensource.org/osd) is based on Debian’s DFSG and [covers essentially the exact same requirements as the Four Freedoms](https://en.wikipedia.org/w/index.php?title=The_Free_Software_Definition&oldid=1244014290#Comparison_with_The_Open_Source_Definition). I think “not open source” is exactly the intended connotation of `lib.licenses.unfree{,Redistributable}`.

Reply:
If we want to be explicit that we’re using a term of art then perhaps we could say “and are not Free and Open Source Software”, but I don’t think that this objection is based on a standard definition of “open source”.

Reply:
There's at least one license that respects user freedom and is (arguably?) not "open source": the SSPL.

I think the wording I put in now is fine too; if the user comes out thinking unfree == "commonly not open source", I think it got the message across.

Reply:
"Open source" is commonly seen as synonymous with FOSS but I don't think that's universal. I would not consider it an esoteric opinion to consider them different. Anyway I'm also fine with the new wording.

> For some context: I only saw this screen in a screenshot and thought to myself: "Hm, that's a little unclear.". I've never used the graphical installer. I can imagine it might perhaps add proprietary drivers if detected though? Perhaps @vlinkz could share some insight here.

I grep'd through the code and did not see any references to any unfree software like nvidia drivers or redistributable firmware or anything else like that. The only reference to this knob in `main.py` is the part that adds `allowUnfree = true;`

https://github.com/NixOS/calamares-nixos-extensions/blob/2d6e267deb8db8d07a7b48047bae72c0000a62b4/modules/nixos/main.py#L296-L299

https://github.com/NixOS/calamares-nixos-extensions/blob/2d6e267deb8db8d07a7b48047bae72c0000a62b4/modules/nixos/main.py#L701-L704

Reply:
actually, there's [a bit that removes unfree kernel module packages from `hardware-configuration.nix`](https://github.com/NixOS/calamares-nixos-extensions/blob/2d6e267deb8db8d07a7b48047bae72c0000a62b4/modules/nixos/main.py#L757-L804) when the unfree box is unchecked. Why on earth? Does `nixos-generate-config` output unfree kernel module packages sometimes?

Uh, yes, it does... Seemingly exclusively for `broadcom_sta` for some reason. That's completely wild.

https://github.com/NixOS/nixpkgs/blob/efc1c323f5c84fd7a21dbf9f7cc264dadbaa8947/nixos/modules/installer/tools/nixos-generate-config.pl#L150-L163

Reply:
Thanks for the investigation! I think the current wording is on point then.

Reply:
I mean personally I'd rather remove the crazy broadcom exclusive treatment from nixos-generate-config, but even still the graphical installer actively removes the unfree package. So the current wording indicates a state this is not desirable and is also not accurate.